observability-lib/api/notification-template.go

```diff
@@ -45,7 +45,7 @@ func (c *Client) DeleteNotificationTemplate(name string) (DeleteNotificationTemp
 	}
 
 	statusCode := resp.StatusCode()
-	if statusCode != 200 {
+	if statusCode != 204 {
```

Comment:
```suggestion
	if statusCode != http.StatusNoContent {
```

Reply:
Great suggestion! Let me fix all the status codes in one go in another PR. cc @Atrax1